src/polkadot/events/balance/index.ts

```diff
@@ -0,0 +1,13 @@
+import {Wallet} from "../../../interfaces";
+import {PolkadotEvent, polkadotEventEmitter} from "../index";
+import {getApi} from "../../api";
+
+export async function registerOnBalanceChange(wallet: Wallet): Promise<void> {
+  const api = await getApi(wallet);
+  const state = wallet.getPluginState();
+  const address = state.polkadot.account.keyring.address;
+  // Here we subscribe to any balance changes and update the on-screen value
+  api.query.system.account(address, ({ data: { free: currentFree }}) => {
+    polkadotEventEmitter.emit(PolkadotEvent.OnBalanceChange, currentFree);
```

Comment:
@mpetrunic Currently, it is actually failing inside this function on `emit` invocation with `Error: serializeObject was passed a object`, I am debugging this right now.

I tested by manually emitting the `OnBalanceChange` event and everything works properly.

Reply:
currentFree.toString() because you cannot serialize bn to pass to frontend